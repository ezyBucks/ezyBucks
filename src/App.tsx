import React, { Component } from 'react';
import './App.css';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import LoginContainer from './Components/Views/Login/LoginContainer';
import RegisterContainer from './Components/Views/Register/RegisterContainer';
import { IndexView } from './Components/Views/Index/IndexView';
import { Header } from './Components/Common/Header';
import { PrivateRoute } from './Components/Common/PrivateRoute';
import { createStore } from 'redux';
import MainReducer from './Redux/authenticate/reducers';
import { Provider, connect } from 'react-redux';
import { AuthenticatedState } from './Redux/authenticate/types';

const store = createStore(MainReducer);

const privateRoute = () => {
    return <div>This is a private route</div>;
};

const publicRoute = () => {
    return <div>This is a public route</div>;
};

class Main extends Component<{ authenticated: any }> {
    render() {
        return (
            <Router>
                <div className="App">
                    <Header />
                    <Route path="/" exact={true} component={IndexView} />
                    <Route path="/login" component={LoginContainer} />
                    <Route path="/register" component={RegisterContainer} />
                    <PrivateRoute
                        path="/private"
                        component={privateRoute}
                        isAuthenticated={this.props.authenticated}
                        redirectPath={'/login'}
                    />
                    <Route path="/public" component={publicRoute} />
                </div>
            </Router>
        );
    }
}

const mapStateToProps = (state: AuthenticatedState) => {
    return {
        authenticated: state.authenticated
    };
};

const MainApp = connect(mapStateToProps)(Main);

export default () => (
    <Provider store={store}>
        <MainApp />
    </Provider>
);
