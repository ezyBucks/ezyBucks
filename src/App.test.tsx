import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import { shallow, mount } from 'enzyme';

it('renders without crashing', () => {
    shallow(<App />);
});

it('fully renders without crashing', () => {
    const wrapper = mount(<App />);
    expect(wrapper.contains(<App />)).toBe(true);
});

it('matches the snapshot', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
});
