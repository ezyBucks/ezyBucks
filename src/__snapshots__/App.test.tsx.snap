// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <span
      style={
        Object {
          "marginRight": "auto",
        }
      }
    >
      ezyBucks
    </span>
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
      type="button"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </button>
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
      type="button"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </button>
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
      type="button"
    >
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </button>
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
      type="button"
    >
      <a
        href="/private"
        onClick={[Function]}
      >
        private
      </a>
    </button>
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
      type="button"
    >
      <a
        href="/public"
        onClick={[Function]}
      >
        public
      </a>
    </button>
  </header>
  <div>
    <div
      className="css-337oej"
    >
      <div
        className="ant-skeleton ant-skeleton-active"
      >
        <div
          className="ant-skeleton-content"
        >
          <h3
            className="ant-skeleton-title"
            style={
              Object {
                "width": "38%",
              }
            }
          />
          <ul
            className="ant-skeleton-paragraph"
          >
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": "61%",
                }
              }
            />
          </ul>
        </div>
      </div>
    </div>
    <div
      className="css-1kztw9e"
    >
      <div
        className="css-dvxtzn"
        style={
          Object {
            "marginTop": "30px",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "15px",
              "width": "400px",
            }
          }
        >
          <h3>
            Transfer
          </h3>
          <div
            className="ant-select ant-select-enabled ant-select-allow-clear"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseOut={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "width": "70%",
              }
            }
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded={false}
              aria-haspopup="true"
              className="ant-select-selection
            ant-select-selection--multiple"
              role="combobox"
            >
              <div
                className="ant-select-selection__rendered"
              >
                <div
                  className="ant-select-selection__placeholder"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "WebkitUserSelect": "none",
                      "display": "block",
                      "userSelect": "none",
                    }
                  }
                  unselectable="on"
                >
                  Select users to transfer to
                </div>
                <ul>
                  <li
                    className="ant-select-search ant-select-search--inline"
                  >
                    <div
                      className="ant-select-search__field__wrap"
                    >
                      <input
                        autoComplete="off"
                        className="ant-select-search__field"
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        value=""
                      />
                      <span
                        className="ant-select-search__field__mirror"
                      >
                        
                        Â 
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingBottom": "15px",
            }
          }
        >
          <h3>
            Amount of ezyBucks to send
          </h3>
          <div
            className="ant-input-number"
            style={Object {}}
          >
            <div
              className="ant-input-number-handler-wrap"
            >
              <span
                aria-disabled={false}
                aria-label="Increase Value"
                className="ant-input-number-handler ant-input-number-handler-up "
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <i
                  aria-label="icon: up"
                  className="anticon anticon-up ant-input-number-handler-up-inner"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="up"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </i>
              </span>
              <span
                aria-disabled={true}
                aria-label="Decrease Value"
                className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <i
                  aria-label="icon: down"
                  className="anticon anticon-down ant-input-number-handler-down-inner"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
            <div
              aria-valuemin={1}
              aria-valuenow={1}
              className="ant-input-number-input-wrap"
              role="spinbutton"
            >
              <input
                autoComplete="off"
                className="ant-input-number-input"
                min={1}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseUp={[Function]}
                required={false}
                step={1}
                value="$ 1"
              />
            </div>
          </div>
        </div>
        <div>
          <button
            className="ant-btn"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            <span>
              Send!
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
